### PR TITLE
Added the models for the search endpoint

### PR DESCRIPTION
**Pull request by cngo-github**
__PR Category__: Feature

Added the intermediary models needed to hold the parameters for the Rosetta `search` endpoints ([link](https://www.rosetta-api.org/docs/SearchApi.html)).

### CODE REVIEW

File: modules/rosetta/src/main/scala/org/tessellation/rosetta/server/model/search/SearchTransactionRequest.scala [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  */
+
+case class SearchTransactionRequest(
+  operation: Operator,
+  address: Option[Address],
+  status: Option[OperationStatus],
+  limit: Option[PosLong],
+  offset: Option[PosLong],
+  transactionHash: Option[Hash],
+  maxBlock: Option[NonNegLong]
```
Comment (marcinwadon) [inline modules/rosetta/src/main/scala/org/tessellation/rosetta/server/model/search/SearchTransactionRequest.scala:19]: What is the maxBlock? Which property of the block it represents?

**Review by marcinwadon — COMMENTED** [review]
Overall looks good. One comment though

**Comment by marcinwadon** [comment]
btw do we need a custom class for search api? Isn't it specified in the openapi thus can be autogenerated?

**Comment by cngo-github** [comment]
> btw do we need a custom class for search api? Isn't it specified in the openapi thus can be autogenerated?

`SearchTransactionRequest` is not the same as the other auto-generated classes. It's purpose is to hold the parameters passed into the [`/search/transactions` endpoint](https://www.rosetta-api.org/docs/SearchApi.html#searchtransactions) and allow those parameters to be passed around in a compact way. It is not a part of the Rosetta specification, so I kept out of the `org.tessellation.rosetta.api` package.

File: modules/rosetta/src/main/scala/org/tessellation/rosetta/server/model/search/SearchTransactionRequest.scala (hunk 1 above)
Comment (cngo-github) [reply modules/rosetta/src/main/scala/org/tessellation/rosetta/server/model/search/SearchTransactionRequest.scala:19]: `maxBlock` is the `max_block` field in the request schema [here](https://www.rosetta-api.org/docs/SearchApi.html#searchtransactions). The endpoint searches for transactions inside of blocks (i.e., snapshots). `maxBlock` specifies the maximum height of the blocks to search.

**Review by marcinwadon — APPROVED** [review]

**Pull request merged**